src/main/java/org/jenkinsci/test/acceptance/docker/fixtures/GitLabContainer.java

```diff
@@ -0,0 +1,204 @@
+package org.jenkinsci.test.acceptance.docker.fixtures;
+
+import org.gitlab4j.api.GitLabApi;
+import org.gitlab4j.api.GitLabApiException;
+import org.gitlab4j.api.GroupApi;
+import org.gitlab4j.api.ProjectApi;
+import org.gitlab4j.api.models.*;
+import org.jenkinsci.test.acceptance.docker.Docker;
+import org.jenkinsci.test.acceptance.docker.DockerContainer;
+import org.jenkinsci.test.acceptance.docker.DockerFixture;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+
+import org.jenkinsci.test.acceptance.po.CapybaraPortingLayer;
+import org.jenkinsci.test.acceptance.utils.ElasticTime;
+
+
+@DockerFixture(id = "gitlab-plugin", ports = {80, 443, 22})
+public class GitLabContainer extends DockerContainer {
+    protected static final String REPO_DIR = "/home/gitlab/gitlabRepo";
+
+    private static final HttpClient client = HttpClient.newBuilder()
+                                                       .followRedirects(HttpClient.Redirect.NORMAL)
+                                                        .connectTimeout(Duration.ofMillis(200))
+                                                       .build();
+
+    private static final ElasticTime time = new ElasticTime();
+
+    public String host() {
+        return ipBound(22);
+    }
+
+    public int sshPort() {
+        return port(22);
+    }
+
+
+    public int httpPort() {
+        return port(80);
+    }
+
+    public String httpHost() {
+        return ipBound(80);
+    }
+
+    public URL getURL() throws IOException {
+        return new URL("http://" + getIpAddress() + sshPort());
+    }
+
+    public URL getHttpUrl() throws IOException {
+        String url = "http://" + httpHost() + ':' + httpPort();
+        return new URL(url);
+    }
+
+    /** URL visible from the host. */
+    public String getRepoUrl() {
+        return "ssh://git@" + host() + ":" + sshPort() + REPO_DIR;
+    }
+
+    public void waitForReady(CapybaraPortingLayer p) {
+        long timeout =  time.seconds(200); // GitLab starts in about 2 minutes add some headway
```

Comment:
this is not actually correct when fed into `waitFor`  as waitFor() creates a `Wait` with an `ElasticTime` as its clock.
so if the factor was 1.5 and you waited 1 second you would end up waiting 2.25 seconds not 1.5 as expected.

https://github.com/jenkinsci/acceptance-test-harness/blob/3bef84db279305367c329dbb06624ac60a6ef3bb/src/main/java/org/jenkinsci/test/acceptance/po/CapybaraPortingLayerImpl.java#L117-L121